vim_onenote.ahk

```diff
@@ -45,65 +49,160 @@ IsLastKey(key)
     return (A_PriorHotkey == key and A_TimeSincePriorHotkey < 400)
 }
 
-; ESC enters Normal Mode
-ESC::
-	Suspend, Off
-    ; send escape on through to exit from find
-    SendInput, {ESC} 
-	ToolTip, OneNote Vim Command Mode Active, 0, 0
-return
+
 
 ;--------------------------------------------------------------------------------
 ; Return to InsertMode
 InsertMode:
     ToolTip
     Suspend, On
+    hotkey, ESC, on
 return
 
 ;--------------------------------------------------------------------------------
+
+;  ctrl + [, ESC enter Normal Mode.
+; Not using ctrl + c, as people may still want to use for copying.
+;^c::
+^[::
+ESC::
+    ; Is not affected by insertmode's suspend
+    suspend, permit
+    gosub NormalMode
+return
+
+; imap workings (eg jj) currently can't be implemented because of how 
+; insert mode works.
+
+
+NormalMode:
+    Suspend, Off
+    ToolTip, OneNote Vim Command Mode Active, 0, 0
+return
+
+
+;--------------------------------------------------------------------------------
 +i::
-    SendInput, {Home}
+    send {Home}
     Gosub InsertMode
 return 
 
 i::Gosub InsertMode
 return 
 
 ;--------------------------------------------------------------------------------
+PrepareClipboard(){
+    ; push clipboard to local variable
+    ClipSaved := ClipboardAll
+    ; Clear clipboard to avoid errors
+    Clipboard :=
+}
+
+Copy(){
+    PrepareClipboard()
+    send ^c
+    ClipWait
+}
+
+Paste(){
+    Send %Clipboard%
+    RestoreClipboard()
+}
+
+RestoreClipboard(){
+    ;restore original clipboard
+    Clipboard := ClipSaved
+    ;ClipWait
+    ClipSaved := ; free memory
+}
+
+GetCursorColumn(){
```

Comment:
The column number? Not currently. 
Adding it to the tooltip might be cool, but it's a complicated method, and currently it reverts any text selection currently in place. It'd have to be updated too frequently for that.

The idea behind it was as a helper method for other things, like a more accurate up/down (see 06c548a52e6bded578be221622ffdda7264518f7).
I wanted to implement it (in conjunction with a row number function, coming later) after reading your comment in the readme.